### PR TITLE
Add custom permissions section to permissions docs

### PR DESCRIPTION
**Pull request by danhayden**
Added information to the permissions section of the docs on adding custom permissions. This information is available in the docs currently but only on the  [FieldPanel docs](https://docs.wagtail.org/en/stable/reference/pages/panels.html#wagtail.admin.panels.FieldPanel.permission) which is not necessarily an obvious place to look.

Also added a note about the issue of creating custom blocks that conflict with the standard model permissions due to the implementation of `format_permissions` https://github.com/wagtail/wagtail/blob/c111deef03c02c24941e06d44274a6ddf83556e4/wagtail/users/templatetags/wagtailusers_tags.py#L70-L71

The problem and discussion that lead to this PR can be found at https://wagtailcms.slack.com/archives/C81FGJR2S/p1666018850881099

### CODE REVIEW

**Comment by danhayden** [comment]
I rendered copy of this change is now available at https://wagtail--9411.org.readthedocs.build/en/9411/topics/permissions.html#displaying-custom-permissions-in-the-admin
I can see there is an issue with the note rendering  `:ref:custom permissions <django:custom-permissions>` that I am not sure how to resolve, I used https://docs.wagtail.org/en/stable/_sources/reference/pages/panels.md.txt as a reference

File: docs/topics/permissions.md [hunk 1]
```diff
@@ -55,6 +55,14 @@ Permission for managing collections themselves can be attached at any point in t
 Users are not allowed to move or delete the collection that is used to assign them permission to manage collections.
 ```
 
+## Adding custom permissions
+
+See Django's documentation on :ref:`custom permissions <django:custom-permissions>` for details on how to set permissions up.
+
+```{note}
+Custom permissions starting with `add_`, `change_` or `delete_` will conflict with the standard model permissions due to how permissions are handled in wagtail.
```
Comment (lb-) [inline docs/topics/permissions.md:63]:
```suggestion
Custom permissions starting with `add_`, `change_` or `delete_` are not currently supported in Wagtail as these will conflict with standard model permissions.
```

Potential reword.

**Comment by danhayden** [comment]
Wording updated and link fixed ([rendered version](https://wagtail--9411.org.readthedocs.build/en/9411/topics/permissions.html#adding-custom-permissions))

Is there anything else needed for this PR?

**Review by lb- — APPROVED** [review]
Changes are looking good now. Thanks!

**Pull request merged**